src/service/url-expander/expander.js

```diff
@@ -298,7 +301,7 @@ export class Expander {
 
         if (val == null) {
           result = '';
-        } else if (this.noEncode_ || NOENCODE_WHITELIST[name]) {
+        } else if (depth !== 0 || this.noEncode_ || NOENCODE_WHITELIST[name]) {
```

Comment:
seems `noEncode_` should be moved out from member field , and get passed around.

Reply:
I kind of like having all the options in one place (the constructor) and only the url in the `expand` call but I don't feel strongly. Again, let me know what you think now that code has changed.